src/app.cpp

```diff
@@ -161,15 +163,37 @@ int App::exec(const Cli& cli) {
         auto beg = high_resolution_clock::now();
 
         for (auto const& input : cli.inputFiles) {
-            max_memory = 0;
+            max_memory = 1024 * 1024 * 50;
+            ExternalResource::get().clear();

```

Comment:
there was a bug when running the compiler with multiple files

Reply:
I don't see how this could bug out other than running out of memory before the preprocessor stage, which I find quite unlikely. Could you provide some steps to reproduce it?

Reply:
could be related to the error I fixed [here](https://github.com/Aerll/rpp/pull/34/commits/247a01a6efaa3683d7a8c42fcf61c1971c5748ab)?

Reply:
I just added `ExternalResource::get().clear();` here, to clear the cache between runs.

`rpp file1.r file2.r` would fail if file1 and file2 include the same file. In that case, the preprocessor would skip the include in the second file. I also normalized the file paths in `ExternalResource`.

There's also a minor issue with circular or transitive inclusion. It goes in infinite recursion instead of halting with a nice error message.

Reply:
[this](https://github.com/Aerll/rpp/pull/34/commits/fb230d46091c0b3ee15e0d66739e9b403351430e) should fix infinite recursion issue. It doesn't give any error, but instead simply refuses to include the same file twice.